modules/related-posts/jetpack-related-posts.php

```diff
@@ -246,6 +251,118 @@ public function get_target_html_unsupported() {
 	}
 
+	/**
+	 * ===============
+	 * GUTENBERG BLOCK
+	 * ===============
+	 */
+
+	/**
+	 * Register the Related Posts Gutenberg block on server.
+	 *
+	 * @action init
+	 * @uses register_block_type
+	 * @return null
+	 */
+	public function register_block() {
+		register_block_type(
+			'a8c/related-posts',
+			array(
+				'render_callback' => array( $this, 'render_block' ),
+			)
+		);
+	}
+
+	/**
+	 * Render the related posts markup.
+	 *
+	 * @return string
+	 */
+	public function render_block( $attributes ) {
+		$block_attributes = array(
+			'show_thumbnails' => isset( $attributes['displayThumbnails'] ) ? (bool) $attributes['displayThumbnails'] : false,
+			'show_date' => isset( $attributes['displayDate'] ) ? (bool) $attributes['displayDate'] : true,
+			'show_context' => isset( $attributes['displayContext'] ) ? (bool) $attributes['displayContext'] : true,
+			'layout' => isset( $attributes['postLayout'] ) && $attributes['postLayout'] === 'list' ? $attributes['postLayout'] : 'grid',
+			'size' => ! empty( $attributes['postsToShow'] ) ? absint( $attributes['postsToShow'] ) : 3,
+		);
+
+		$related_posts = $this->get_for_post_id( get_the_ID(), array(
+			'size' => $block_attributes['size'],
+		) );
+
+		if ( ! $related_posts ) {
+			return '';
+		}
+
+		ob_start();
+		?>
+		<div id="jp-relatedposts" class="jp-relatedposts" style="display: block;">
+			<div class="jp-relatedposts-items jp-relatedposts-items-visual jp-relatedposts-<?php echo esc_attr( $block_attributes['layout'] ); ?>">
+				<?php foreach ( $related_posts as $index => $related_post ): 
+					$classes = array_filter( array(
+						'jp-relatedposts-post',
+						'jp-relatedposts-post' . $index,
+						! empty( $block_attributes['show_thumbnails'] ) ? 'jp-relatedposts-post-thumbs' : '',
+					) );
+					?>
+					<div
+						class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>"
+						data-post-id="<?php echo esc_attr( $related_post['id'] ); ?>"
+						data-post-format="<?php echo esc_attr( ! empty( $related_post['format'] ) ? $related_post['format'] : 'false' ); ?>"
+					>
+						<?php if ( ! empty( $block_attributes['show_thumbnails'] ) && ! empty( $related_post['img']['src'] ) ): ?>
+							<a class="jp-relatedposts-post-a"
+								href="<?php echo esc_url( $related_post['url'] ); ?>"
+								title="<?php echo esc_attr( $related_post['title'] ); ?>"
+								rel="<?php echo esc_attr( $related_post['rel'] ); ?>"
+								data-origin="<?php echo esc_attr( $related_post['url_meta']['origin'] ); ?>"
+								data-position="<?php echo esc_attr( $related_post['url_meta']['position'] ); ?>"
+							>
+								<img class="jp-relatedposts-post-img"
+									src="<?php echo esc_url( $related_post['img']['src'] ); ?>"
+									width="<?php echo esc_attr( $related_post['img']['width'] ); ?>"
+									alt="<?php echo esc_attr( $related_post['title'] ); ?>"
+								/>
+							</a>
+						<?php endif; ?>
+
+						<h4 class="jp-relatedposts-post-title">
```

Comment:
This is no longer relevant since we don't have options the heading in the block itself, in favor of the separate Heading block. See https://github.com/Automattic/wp-calypso/pull/27454 for context and rationale.